src/prefect/task_engine.py

```diff
@@ -622,21 +623,24 @@ def setup_run_context(self, client: Optional[SyncPrefectClient] = None):
 
             self.logger = task_run_logger(task_run=self.task_run, task=self.task)  # type: ignore
 
-            if not PREFECT_EXPERIMENTAL_ENABLE_CLIENT_SIDE_TASK_ORCHESTRATION:
-                # update the task run name if necessary
-                if not self._task_name_set and self.task.task_run_name:
-                    task_run_name = _resolve_custom_task_run_name(
-                        task=self.task, parameters=self.parameters
-                    )
+            # update the task run name if necessary
+            if not self._task_name_set and self.task.task_run_name:
```

Comment:
This and below is me reverting a change I made to mode setting the task run name in `initialize_run`. I realize now this was a byproduct of how I had initially setup tests before we were using the `events_pipeline`.

Leaving this here lets do things like set the task run name based on the task run context.